server/adaptors/integrations/repository/fs_data_adaptor.ts

```diff
@@ -48,6 +48,15 @@ function tryParseNDJson(content: string): object[] | null {
   }
 }
 
+// Check if a location is a directory without an exception if location not found
+const safeIsDirectory = async (maybeDirectory: string): Promise<boolean> => {
+  try {
+    return (await fs.lstat(maybeDirectory)).isDirectory();
+  } catch (_err: unknown) {
+    return false;
+  }
+};
+
```

Comment:
This change is already part of: https://github.com/opensearch-project/dashboards-observability/pull/1155 do you still want to add this here?

Reply:
This PR depends on #1155, the true diff is about 30 lines.